mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -558,6 +475,7 @@ object KMeans {
    * Trains a k-means model using specified parameters and the default values for unspecified.
    */
   @Since("0.8.0")
+  @deprecated("Use train method without 'runs'", "2.1.0")
```

Comment:
Yes, though there's no alternative to those with the same arguments. We could add another overload and deprecate the others. I'm OK with that too, just felt a little gross to add yet more.